invenio_records_rest/facets.py

```diff
@@ -38,6 +39,42 @@ def inner(values):
     return inner
 
 
+def range_filter(field, **kwargs):
+    """Create a range filter."""
+    start_date_math = None
+    end_date_math = None
+    args = kwargs.copy()
+
+    if 'start_date_math' in kwargs:
+        start_date_math = '||' + args['start_date_math']
+        del args['start_date_math']
+
+    if 'end_date_math' in kwargs:
+        end_date_math = '||' + args['end_date_math']
+        del args['end_date_math']
+
+    def inner(values):
+        [start, end] = values[0].split('--')
```

Comment:
If no '--' is found, the terms_filter will be now used instead, as I believe 1940- may be a valid value in some context.